### PR TITLE
Update coupon admin to be store aware

### PR DESCRIPTION
**Pull request by RyanMacG**

### CODE REVIEW

File: app/controllers/concerns/spree_multi_domain/promotions_by_store_admin.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module SpreeMultiDomain::PromotionsByStoreAdmin
+  extend ActiveSupport::Concern
+
+  def self.prepended(base)
```
Comment (davidwinter) [inline app/controllers/concerns/spree_multi_domain/promotions_by_store_admin.rb]: You should be able to use `included do` to save on some of this stuff.

File: app/models/spree/promotion_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Spree::Promotion.class_eval do
+  belongs_to :store
```
Comment (davidwinter) [inline app/models/spree/promotion_decorator.rb:2]: Can we not use a concern?

Comment (RyanMacG) [reply app/models/spree/promotion_decorator.rb:2]: Was following the existing structure of the other models, but we might be able to

**Comment by RyanMacG** [comment]
@SebAshton if we include these rather than using prepend it the before_filter fails to work correctly

**Comment by SebAshton** [comment]
If that's the case, stick with the prepend. Going to throw this out there too...

```
def index
    merge_store_id_query

    super
end
```

will potentially side step any before_filter related pain.

**Pull request closed without merging**